hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -169,6 +178,8 @@ private void assignInserts(WorkloadProfile profile, HoodieEngineContext context)
 
     for (String partitionPath : partitionPaths) {
       WorkloadStat pStat = profile.getWorkloadStat(partitionPath);
+      WorkloadStat executionStat = executionProfile.getOrDefault(partitionPath, new WorkloadStat());
+      executionProfile.putIfAbsent(partitionPath, executionStat);
```

Comment:
should we move this to L244 or something. i.e. after updating executionStat and then update executionProfile map.